### ISSUE TITLE
Erstellung der JWS-Signatur - Beispiel

### ISSUE THREAD

**Issue by MichaelSchustereder** [a-sit-plus/at-registrierkassen-mustercode#262]
Hallo,

ich versuche das Beispiel aus dem Dokument Detailfragen 1.2 von Seite 43 für die Berechnung des Payload nachzuvollziehen. Hier werden folgende Belegdaten angeführt:

_R1-AT0_DEMO-CASH-BOX524_366588_2015-12-17T11:23:44_0,00_0,00_0,00_26,05_0,00_m8LGyyY4UAA=_20f2ed172daa09e5_5HjRCx+XIz4=

Laut Beispiel ergibt das BASE64-Url kodiert folgenden Wert:

X1IxLUFUMF9ERU1PLUNBU0gtQk9YNTI0XzM2NjU5Nl8yMDE1LTEyLTE3VDExOjIzOjQ0XzAsMDBfMCwwMF8zLDY0Xy0yLDYwXzEsNzlfVkZKQl80N2JlNzM3Y2IxZjZkMWYxX1p2TnhKdzZhMUE0PQ

Ich komme einfach auf den Wert nicht hin. Mein errechneter Wert ist und bleibt 

X1IxLUFUMF9ERU1PLUNBU0gtQk9YNTI0XzM2NjU4OF8yMDE1LTEyLTE3VDExOjIzOjQ0XzAsMDBfMCwwMF8wLDAwXzI2LDA1XzAsMDBfbThMR3l5WTRVQUE9XzIwZjJlZDE3MmRhYTA5ZTVfNUhqUkN4K1hJejQ9

Kann mir bitte jemand das Beispiel überprüfen und bescheid geben, ob das angeführte Beispiel in den Detailfragen auch korrekt ist? Ich bin mir mittlerweilen nicht mehr sicher.

Herzlichen Dank!

LG, Michael Schustereder

**Comment by asitplus-pteufl** [comment]
Haben Sie die Signatur mitkodiert? die gehört in die JWS-Darstellung nicht mehr rein?

Am besten ist es sie nehmen die fertigen Beispiele die mit dem Mustercode geliefert werden (oder selbst über die DemoKasse über die CMD-Line erstellt werden können).

**Comment by MichaelSchustereder** [comment]
Danke für die rasche Rückmeldung! Nein, die Signatur habe ich nicht mitkodiert. Ich habe nur den Wert, wie oben angeführt, kodiert.

Meinen sie die die Beispiele unter at-registrierkassen-mustercode/regkassen-democashbox/src/main/resources/TESTSUITE_TEST_SZENARIO_1.json? Falls ja, dann ich in den Beispielen leider nicht erkennen, ob meine Kodierung des payload funktioniert.

**Comment by asitplus-pteufl** [comment]
Nein, diese Datei nicht wenn sie Unter Releases das Projekt herunterladen, werden auch die DEMO-Belege zur Verfügung gestellt. Dort sehen sie die von der DemoKasse erstellten Belege die nach den von Ihnen angesprochenen Testszenarien erstellt wurden, da gibt es einmal das DEP-Format und einmal in einer Datei wo die rohen Belege gespeichert sind. Da ist es besser die Beispiele nachzustellen, bei den Beispielen in der Detailspezifikation kann es durchaus vorkommen, dass Werte nur zu repräsentativen Zwecken enthalten sind und nicht notwendigerweise korrekt abgebildet sind.

**Comment by asitplus-pteufl** [comment]
Hier der direkte Link zu den BSP-Belegen: https://github.com/a-sit-plus/at-registrierkassen-mustercode/releases/download/V1.0.0/example-output-1.0.0.zip

**Comment by MichaelSchustereder** [comment]
Ich habe nun mit diesen Beispielen verglichen und meine Routine funktioniert. Es ist offenbar tatsächlich so, dass in den Detailfragen die Beispiele nicht korrekt abgebildet sind. Leider fehlt der Hinweis, dass diese mitunter nur repräsentative Zwecke darstellen.

Danke nochmals für die äußerst rasche Hilfe!

**Comment by asitplus-pteufl** [comment]
Ich nehme den Hinweis entgegen. Lasse daher das Issues offen. Freut mich dass es jetzt passt.
Ein Hinweis noch: BASE64-URL erlaubt das Padding Zeichen =. Die JWS-Spezfiikation schließt dieses explizit aus, d.h. alle = aus der Kodierung entfernen.

**Comment by simonmau** [comment]
Könnte man das vielleicht als bekannten Fehler vermerken (ungefähr [hier](https://github.com/a-sit-plus/at-registrierkassen-mustercode#festlegungen-des-bmf-zu-detailfragen-der-registrierkassensicherheitsverordnung-rksv))? Ansonsten suchen noch weitere verzweifelte Leute nach einem Fehler, wo gar kein Fehler ist.
